src/models/types/circuit/circCompDot.mc

```diff
@@ -0,0 +1,108 @@
+        <table BORDER=",quote,"0",quote," CELLBORDER=",quote,"0",quote," CELLSPACING=",quote,"0",quote," CELLPADDING=",quote,"0",quote,"> 
+            <tr>",
+                (foldl (lam str. lam x. concat str (makeTDElem x.0 x.1 x.2 quote))) "" 
+                    [
+                        ("black",side_width,side_height),
+                    ("none",center_width,side_height),
+                    ("none",(addi side_width 1),side_height)],
+            "</tr> 
+            <tr>",
+                (foldl (lam str. lam x. concat str (makeTDElem x.0 x.1 x.2 quote))) "" 
+                    [("black",side_width,side_height),("none",center_width,center_height),("black",(addi side_width 1),side_height)],
+            "</tr>
+            <tr>",
+                (foldl (lam str. lam x. concat str (makeTDElem x.0 x.1 x.2 quote))) "" 
+                    [("black",side_width,side_height), ("none",center_width,side_height),("none",(addi side_width 1),side_height)],
+            "</tr>   
+        </table>>"
+    ] in (setting,"V") in
+    formatComponentToDot name quote (concatList ["xlabel=",quote,value,settings.1,quote," ",settings.0]) isConnected
+
+-- returns the ground component in dot.
+let circGroundToDot = lam quote. lam name. lam custom_settings. lam isConnected.
+    let width =5 in
+    let height = 1 in
+    let settings = match custom_settings with Some (setting,unit) then (setting,unit) else
+        (foldl concat [] ["shape=none, color=none height=0 width=0 margin=0 label=<
+    <table CELLBORDER=",quote,"0",quote," CELLSPACING=",quote,"0",quote," CELLPADDING=",quote,"0",quote," >\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["black","black","black","black","black"],
+        " </tr>\n<tr>",
+           makeTDElem "none" width (muli 2 height) quote,
+       "</tr>\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["none","black","black","black","none"],
+        "</tr>\n<tr>",
+            makeTDElem "none" width (muli 2 height) quote,
+        "</tr>\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["none","none","black","none","none"],
+        "</tr>\n</table>> "]
+        ,"") in
+    formatComponentToDot name quote (concatList ["label=",quote,quote,settings.0]) isConnected
+
+-- returns a custom component in dot.
+let circOtherToDot = lam quote. lam name. lam value. lam _. lam custom_settings. lam isConnected.
+    let settings = match custom_settings with Some (setting,unit) then (setting,unit) else
+        (foldl concat [] ["style=filled fillcolor=white shape=circle label=",quote,quote, " xlabel=",quote,value,quote]," ") in
+    let value_str = match value with "" then "" else (foldl concat [] [value," ",settings.1," "]) in
+    formatComponentToDot name quote (concatList ["xlabel=",quote,value_str,quote," ",settings.0]) isConnected
```

Comment:
You can change these `foldl concat []` to `join`

Reply:
Fixed!